compiler/rustc_parse/src/parser/nonterminal.rs

```diff
@@ -44,6 +44,8 @@ impl<'a> Parser<'a> {
                 && !token.is_keyword(kw::Let)
                 // This exception is here for backwards compatibility.
                 && !token.is_keyword(kw::Const)
+                // FIXME(vincenzopalazzo): query less time the edition
+                && (token.span.edition().at_least_rust_2024() && !token.is_keyword(kw::Underscore) )
```

Comment:
Checking the span is not necessary since #126700

Reply:
yeah I think this may be also the reason of the compilation error, thanks for looking into it